cmd/clusterctl/client/cluster/mover.go

```diff
@@ -611,6 +644,38 @@ func setClusterClassPause(ctx context.Context, proxy Proxy, clusterclasses []*no
 	return nil
 }
 
+// checkClustersNotPaused checks that no cluster in the graph is paused before proceeding.
+func checkClustersNotPaused(ctx context.Context, proxy Proxy, clusters []*node) error {
+	for _, cluster := range clusters {
+		clusterObj := &clusterv1.Cluster{}
+		if err := getClusterObj(ctx, proxy, cluster, clusterObj); err != nil {
+			return err
+		}
+
+		if ptr.Deref(clusterObj.Spec.Paused, false) || annotations.HasPaused(clusterObj) {
+			return errors.Errorf("cannot start operation while Cluster %s/%s is paused", clusterObj.Namespace, clusterObj.Name)
+		}
+	}
```

Comment:
What about collecting a list of paused clusters and returning error only with there is at least one paused cluster / with a message including the list of paused clusters
(if we are failing when we met the first paused cluster, the user doesn't have an easy way to figure it out if other clusters are in the same state)

Same for checkClusterClassesNotPaused